### ISSUE TITLE
flatline rule does not fire correctly

### ISSUE THREAD

**Issue by anishmashankar** [Yelp/elastalert#822]
Here is a configuration for a flatline alert I set up
`
name: flatline check

index: logstash-*

type: flatline

timeframe:
  minutes: 15

threshold: 1

es_host: localhost
es_port: 9200

realert:
  minutes: 0

query_delay:
  minutes: 15

filter:
- query:
    query_string:
      query: "name: app-name"

query_key:
  - env
  - property

alert:
  - slack

alert_text: |
  {0}
  remediations: restart app

alert_text_args:
  - key

alert_subject: |
  the app flatlined
`
When I test this rule, i get the following output in the alerts section

`
INFO:elastalert:Alert for App flatline check at 2016-11-22T07:40:41.276169Z:
INFO:elastalert:production, property
remediations: restart app

An abnormally low number of events occurred around 2016-11-22 07:40 UTC.
Between 2016-11-22 07:25 UTC and 2016-11-22 07:40 UTC, there were less than 1 events.

@timestamp: 2016-11-22T07:40:41.276169Z
count: 0
key: production,property
`

I got tons of these messages on the console. It is observed that the alert did not fire correctly. The flatline rule alerted even though documents for the time range existed on elasticsearch. If realert is set to 15 minutes, only one alert fires even if keys for the next alert are different.

Elastalert 0.1.4.
Am I doing something wrong?

**Comment by Qmando** [comment]
That indicates that a flatline occurred for documents matching ``env: production AND property: property``. Can you find a document that matches that that occurred in the 15 minutes.

You can add this patch if you want some additional debug information. It will show you all of the different (env, property) combinations that elastalert has seen at the end of each query cycle.
```
diff --git a/elastalert/ruletypes.py b/elastalert/ruletypes.py
index f53d2f4..42c81e2 100644
--- a/elastalert/ruletypes.py
+++ b/elastalert/ruletypes.py
@@ -503,6 +503,7 @@ class FlatlineRule(FrequencyRule):
     def garbage_collect(self, ts):
         # We add an event with a count of zero to the EventWindow for each key. This will cause the EventWindow
         # to remove events that occurred more than one `timeframe` ago, and call onRemoved on them.
+        print self.occurrences.keys()
         default = ['all'] if 'query_key' not in self.rules else []
         for key in self.occurrences.keys() or default:
```

**Comment by anishmashankar** [comment]
Yes,  @Qmando. I see there are documents in elasticsearch for that comnbination of env and property in that time interval

**Comment by sivatarunp** [comment]
any update on this @Qmando facing same issue

**Comment by Qmando** [comment]
@sivatarunp  I don't think this is an issue. If it was, it's likely been fixed. If you believe it is, please post your config, the alert, and the full source of a document you believe was missed/ignored.
